### DISCUSSION TITLE
Extension Support for Suites

### DISCUSSION THREAD

**Discussion by khmarbaise** [junit-team/junit-framework#4157] [General]
At the moment you can intercept etc. all the parts of the life cycle of tests via the extension mechnism very nicely (for example `BeforeTestExecutionCallback`etc.), but unfortunately the existing suite parts do not have such support yet?

**Comment by sbrannen** [comment]
The test lifecycle extension APIs you are referring to belong to the JUnit Jupiter extension model.

Whereas, `@BeforeSuite` and `@AfterSuite` are features of the JUnit Platform Suite Engine, and the suite engine does not have an extension model.

And... there is no cross-pollination between the Jupiter and Suite engines.

In other words, Jupiter doesn't know it's being run in a suite, and the suite engine does not know anything about (or have any mechanism to integrate with) the Jupiter extension model.

Hopefully that answers your question.

Though, you may also be "proposing something in the form of a question". 😉 

If so, what exactly are you proposing?

**Comment by marcphilipp** [comment]
Just thinking out loud, #2816 would allow creating resources in `@BeforeSuite` methods (via some kind of parameter) that engines could access if they support `NamespacedHierarchicalStore`. The Suite Engine could create a `NamespacedHierarchicalStore` for each `@Suite` class (plus one on the engine level to allow reuse when there are multiple `@Suite` classes). Whether or not it would directly expose `NamespacedHierarchicalStore` or some simplified decorator of it would have to be discussed.
